reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -123,6 +123,30 @@ bool ReconfigurationHandler::handle(const concord::messages::RestartCommand& com
   handleWedgeCommands(command.bft_support, true, command.restart, true);
   return true;
 }
+bool ReconfigurationHandler::handle(const concord::messages::InstallCommand& cmd,
+                                    uint64_t sequence_num,
+                                    uint32_t,
+                                    concord::messages::ReconfigurationResponse& rres) {
+  concord::messages::ReconfigurationErrorMsg error_msg;
+  if (cmd.version.empty()) {
+    LOG_ERROR(getLogger(), "InstallCommand received with empty version string at seq_num " << sequence_num);
+    return false;
+  }
+  LOG_INFO(getLogger(), "InstallCommand instructs replica to stop at seq_num " << KVLOG(sequence_num, cmd.version));
+  bftEngine::ControlStateManager::instance().setStopAtNextCheckpoint(sequence_num);
+  // TODO(NK): set remove_metadata and unwedge flag to True once we support start new epoch with (n-f) nodes
```

Comment:
If we don't remove the metadata on the new software install, it means that our new SW version has to be backward compatible (at least with the metadata scheme).
I wonder if it wouldn't be better to remove the metadata and state that we don't support n-f recovery (as with add/remove). WDYT?

As for the new epoch, in my opinion we need to start a new epoch even if don't remove the metadata

Reply:
Eventually, we need to remove metadata, when we remove the condition to start the cluster only when all n nodes are working. 
Right now, i am assuming that metadata will be compatible with new version of software. If we state that we support (n-f) install only (like add/remove) then we are mandating customer to spin a new VM and add that to the cluster (which also means, reconfiguration step post install). I feel, in first phase we can allow cluster to operate even after (n-f) install (with assumption that metadata is compatible with new software). Do you see any issue with this implementation?

Reply:
no, its ok.
I would only add a new epoch here (you can see how it is done in the unwedge handler without restarting concordbft, simply add a new block with the new epoch and continue running)